src/wp-admin/options-general.php

```diff
@@ -107,34 +107,78 @@
-		$classes_for_avatar          .= ' has-site-icon';
+		$classes_for_wrapper         .= ' has-site-icon';
 		$classes_for_button           = $classes_for_update_button;
 		$classes_for_button_on_change = $classes_for_upload_button;
 	} else {
-		$classes_for_avatar          .= ' hidden';
+		$classes_for_wrapper         .= ' hidden';
 		$classes_for_button           = $classes_for_upload_button;
 		$classes_for_button_on_change = $classes_for_update_button;
 	}
 
+	// Handle alt text for site icon on page load.
+	$site_icon_id           = (int) get_option( 'site_icon' );
+	$app_icon_alt_value     = '';
+	$browser_icon_alt_value = '';
+
+	if ( $site_icon_id ) {
+		$img_alt            = get_post_meta( $site_icon_id, '_wp_attachment_image_alt', true );
+		$filename           = wp_basename( get_site_icon_url() );
+		$app_icon_alt_value = sprintf(
+			/* Translators: %s: The selected image filename. */
+			__( 'App icon preview: The current image has no alternative text. The file name is: %s' ),
+			$filename
+		);
+
+		$browser_icon_alt_value = sprintf(
+			/* Translators: %s: The selected image filename. */
+			__( 'Browser icon preview: The current image has no alternative text. The file name is: %s' ),
+			$filename
+		);
 
+		if ( $img_alt ) {
+			$app_icon_alt_value = sprintf(
+				/* Translators: %s: The selected image alt text. */
+				__( 'App icon preview: Current image: %s' ),
+				$img_alt
+			);
+
+			$browser_icon_alt_value = sprintf(
+				/* Translators: %s: The selected image alt text. */
+				__( 'Browser icon preview: Current image: %s' ),
+				$img_alt
+			);
+		}
+	}
 	?>
-	<div id="site-icon-preview" class="site-icon-preview wp-clearfix <?php echo esc_attr( $classes_for_avatar ); ?>">
-		<div class="favicon-preview">
-			<img src="<?php echo esc_url( admin_url( 'images/' . ( is_rtl() ? 'browser-rtl.png' : 'browser.png' ) ) ); ?>" class="browser-preview" width="182" alt="">
-			<div class="favicon">
-				<img src="<?php site_icon_url(); ?>" alt="<?php esc_attr_e( 'Preview as a browser icon' ); ?>">
-			</div>
-			<span class="browser-title" aria-hidden="true"><?php echo get_bloginfo( 'name' ); ?></span>
+
+<style>
+:root{
+	--site-icon-url: url( '<?php site_icon_url(); ?>' );
+}
+</style>
+
+<div id="site-icon-preview" class="site-icon-preview-inline settings <?php echo esc_attr( $classes_for_wrapper ); ?>">
+	<img src="<?php site_icon_url(); ?>" id="app-icon-preview" class="home-icon" alt="<?php echo esc_attr( $app_icon_alt_value ); ?>"/>
+	<div class="site-icon-preview-browser">
+		<svg role="img" aria-label="" width="48" height="40" fill="none" xmlns="http://www.w3.org/2000/svg" class="site-icon-preview-browser-buttons"><path fill-rule="evenodd" clip-rule="evenodd" d="M0 20a6 6 0 1 1 12 0 6 6 0 0 1-12 0Zm18 0a6 6 0 1 1 12 0 6 6 0 0 1-12 0Zm24-6a6 6 0 1 0 0 12 6 6 0 0 0 0-12Z" /></svg>
+		<div class="site-icon-preview-tab">
+			<img src="<?php site_icon_url(); ?>" class="site-icon" id="browser-icon-preview" />
```

Comment:
This image tag is now missing the alt text specified in `$browser_icon_alt_value`.